pkg/strc/middleware_logger.go

```diff
@@ -102,3 +102,92 @@ func EchoRequestLogger(logger *slog.Logger, config MiddlewareConfig) echo.Middle
+// the context. If the request does not have a trace id, a new one is generated. This feature can be
+// disabled.
+//
+// Finally, it can be configured to dump request and response bodies, request and response headers,
+// and user agent.
+//
+// Requests with errors are logged using slog.Error().
+// Requests without errors are logged using slog.Info().
+func RequestLogger(logger *slog.Logger) func(http.Handler) http.Handler {
+	return RequestLoggerWithConfig(logger, MiddlewareConfig{
+		DefaultLevel:     slog.LevelInfo,
+		ClientErrorLevel: slog.LevelWarn,
+		ServerErrorLevel: slog.LevelError,
+		SpanName:         "http request",
+		Filters:          []Filter{},
+	})
+}
+
+// RequestLoggerWithFilters returns a `func(http.Handler) http.Handler` (middleware) that logs requests using slog.
+//
+// Requests with errors are logged using slog.Error().
+// Requests without errors are logged using slog.Info().
+func RequestLoggerWithFilters(logger *slog.Logger, filters ...Filter) func(http.Handler) http.Handler {
+	return RequestLoggerWithConfig(logger, MiddlewareConfig{
+		DefaultLevel:     slog.LevelInfo,
+		ClientErrorLevel: slog.LevelWarn,
+		ServerErrorLevel: slog.LevelError,
+		SpanName:         "http request",
+		Filters:          filters,
+	})
+}
+
+// RequestLoggerWithConfig returns a `func(http.Handler) http.Handler` (middleware) that logs requests using slog.
+func RequestLoggerWithConfig(logger *slog.Logger, config MiddlewareConfig) func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			for _, filter := range config.Filters {
+				if !filter(nil, r) {
+					next.ServeHTTP(w, r)
+					return
+				}
+			}
+
+			start := time.Now()
+
+			bw := newBodyWriter(w, ResponseBodyMaxSize, config.WithResponseBody)
+			next.ServeHTTP(bw, r)
+
+			end := time.Now()
+			latency := end.Sub(start)
+
+			// errors should be handled by echo.HTTPErrorHAndler, but let's grab the status at least.
+			// Anthing else should be handled by the error handler.
+			attrs := []slog.Attr{
+				slog.Time("time", start.UTC()),
+				slog.Duration("latency", latency),
+			}
+			attrs = append(attrs, slogAttributesFromRequest(r)...)
+
+			status := bw.status
+			attrs = append(
+				attrs,
+				slog.Attr{
+					Key: "response",
+					Value: slog.GroupValue(
+						[]slog.Attr{
+							slog.Int64("length", int64(bw.bytes)),
+							slog.Int("status", status),
+						}...,
+					),
+				},
+			)
+
+			level := config.DefaultLevel
+			if status >= http.StatusInternalServerError {
+				level = config.ServerErrorLevel
+				attrs = append(attrs, slog.String("error", bw.body.String()))
+			} else if status >= http.StatusBadRequest && status < http.StatusInternalServerError {
+				level = config.ClientErrorLevel
+			}
+
+			logger.LogAttrs(r.Context(), level, fmt.Sprintf("%d: %s", status, http.StatusText(status)), attrs...)
+		})
+	}
+}
```

Comment:
I think we can drop stdlib middleware for now completely. But if you want to keep it since the work is done, sure.